clorm/orm/core.py

```diff
@@ -2817,8 +2924,8 @@ class Booking(Predicate):
     The constructor creates a predicate instance (i.e., a *fact*) or complex
     term.
 
-    Note: Using the ``raw`` parameter is now deprecated. You should use the
-    `unify()` function or `Unifier` class instead.
+    Note: Using the ``raw`` parameter is no longer supported. You should use
+    the `unify()` function or `Unifier` class instead.
 
     Args:
```

Comment:
Can we add `Predicate._unify` if TYPE_CHECKING is True (see little down below like we do for `_meta` and `_field`)? Because it is dynamically created it would be otherwise unkown to static type-checkers...

```python
if typing.TYPE_CHECKING:
    @classmethod
    def _unify(cls: Type[_P], raw: AnySymbol) -> Optional[_P]:
        pass
```

Reply:
Ok, I'll add it.  But, is it also possible to do something similar for the __init__() ? I'm just not sure how to specify some meaningful arguments since the only two arguments that we know are the `sign` and `raw` keyword arguments. So, the following doesn't look right but its the best I could come up with:

``` 
if typing.TYPE_CHECKING:
        def __init__(self: Type[_P], *args: Tuple[_,...], **kwargs: Dict[str, _])-> None:
            pass
```

Reply:
For `__init__` we don't need this (it actually causes more problems^^). If no initializer is defined, type-checkers always fall back to the default one `__init__(self, *args, **kwargs)`
A more elaborated version we could get with #80. It should be quite easy to implement

Reply:
Ok. I'll remove it. I just accepted your PR which add the `__dataclass_transform__` stuff so that will fix the issue :+1: